designs/2023-rule-performance-dashboard/README.md

```diff
@@ -0,0 +1,276 @@
+- Repo: eslint/eslint
+- Start Date: 2023-03-10
+- RFC PR: <https://github.com/eslint/rfcs/pull/108>
+- Authors: Mara Nikola Kiefer (@mnkiefer)
+
+# Performance Statistics
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This document describes a new ESLint option `stats`, which enables the user to obtain a series of runtime *statistics* on top of their final lint results.
+
+These statistics include more granular information, such as the [parse](#timing-parse)-, [fix](#timing-fix)-, and [lint-times](#timing-lint) and the number of [fix passes](#fix-passes), [directives and violations](#directives-and-violations) encountered.
+
+| Stats |  | Description |
+|--------|-------|---|
+| **timing** | total  | The overall lint time that is spent on a file. This includes `parse-`, `fix-` and all `rules-` (total) times. |
+|        | parse | The time that is spent when parsing a file, that is when [`parse()`](#timing-parse) or `parseForEslint()` is called. |
+|        | rules | This is similar to the rule performance TIMING, except that it is more granular, that is, persisted as *per file per rule*.  |
+| **passTiming** | | An array of *timing* objects for each of the [fix passes](#fix-passes). In total, these amount to the (total) *timing* object for the respective file. |
+| **fixPasses** | | The number of [fixes](https://eslint.org/docs/latest/use/command-line-interface#fix-problems) successfully applied to a file. |
+| **directives** | | Refer to [this issue comment](https://github.com/eslint/eslint/issues/14597#issuecomment-1003863524) |
```

Comment:
In order to implement it, I need to better understand how this different from `suppressMessages` on the result object (see enquiry by https://github.com/eslint/eslint/issues/14597#issuecomment-1003722176).
```md
 * @property {SuppressedLintMessage[]} suppressedMessages All of the suppressed messages for the result.
```
Could you please elaborate on that?

Reply:
I'm not exactly familiar with what is represented by that property.

But at a conceptual level, doesn't it make sense to include the count of directive comments affecting a rule, as well as a count of the violations actually suppressed for the rule? Violations could be suppressed for various reasons. A single directive comment could cause multiple suppressions (if used to disable a rule for an entire file, for example).

Reply:
I'd like to avoid scope creep in this RFC -- we don't need to include every possible metric out of the box. Let's just focus on returning the data from rule runs to match `TIMING=1` in this RFC and we can always talk enhancements for additional data points to include later. The design of this RFC was changed intentionally to allow for additional pieces of data to be added later on, so I don't want to block on which data to include that we don't already expose.

Reply:
Agreed that we don't need to include all possible metrics at this point. I am happy that the current design includes a few initial metrics and can be expanded to add more later.

My underlying goal with this comment thread is just to better understand how we are choosing the initial set of metrics and ensure we've considered the ones I proposed in [this](https://github.com/eslint/eslint/issues/14597#issuecomment-1003863524) comment. I understand if we can't or don't want to include all of them. If some metrics are out of scope for now but would be fair game for follow-up later, that would also be helpful to know.

Your comment does help answer some of this:

> Let's just focus on returning the data from rule runs to match TIMING=1 in this RFC...